code/menuui/playermenu.cpp

```diff
@@ -334,8 +334,9 @@ void player_select_init()
 	} else { // otherwise go to the single player mode by default
 		player_select_init_player_stuff(PLAYER_SELECT_MODE_SINGLE);
 	}
-
-	if ( (Player_select_num_pilots == 1) && Player_select_input_mode ) {
```

Comment:
The code was logically dead because `Player_select_input_mode` couldn't be true at that point.